src/main/scala/com.snowplowanalytics/weather/providers/openweather/OwmAsyncClient.scala

```diff
@@ -13,46 +13,80 @@
 package com.snowplowanalytics.weather
 package providers.openweather
 
-// Scala
-import scala.concurrent.Future
-import scala.concurrent.ExecutionContext.Implicits.global
+// cats
+import cats.effect.Sync
+import cats.implicits._
+
+// circe
+import io.circe.parser.parse
+import io.circe.{Decoder, Json}
+
+// hammock
+import hammock.{Hammock, HttpResponse, Method, Status, Uri}
+import hammock.jvm.Interpreter
 
 // This library
-import Errors.WeatherError
+import Errors._
 import Responses._
 import Requests._
 
 /**
  * Asynchronous OpenWeatherMap client
  *
  * @param appId API key
- * @param transport HTTP client for send requests, receive responses
  */
-class OwmAsyncClient(appId: String, transport: HttpAsyncTransport) extends Client[AsyncWeather] {
-  def receive[W <: OwmResponse: Manifest](request: OwmRequest): Future[Either[WeatherError, W]] = {
-    val processedResponse = transport.getData(request, appId)
-    processedResponse.map(_.right.flatMap(extractWeather[W]))
-  }
-}
+case class OwmAsyncClient[F[_]: Sync](appId: String, apiHost: String = "api.openweathermap.org", ssl: Boolean = false)
+    extends Client[F] {
 
-/**
- * Companion object for async client
- */
-object OwmAsyncClient {
+  private implicit val interpreter = Interpreter[F]
+
+  def receive[W <: OwmResponse: Decoder](request: OwmRequest): F[Either[WeatherError, W]] = {
+
+    val scheme = if (ssl) "https" else "http"
+
+    val uriOrError = Uri
+      .fromString(s"$scheme://$apiHost")
```

Comment:
can we construct this uri using the dsl? this way we won't have to deal with an error in uri creation